{doc} Add breaking change rule doc (#26779)
Azure/azure-cli: doc/breaking_change_rules/1004.md

### 1004 - Cmd Property Removed

**Description**: Checks whether a new property of a cmd is removed from the previous cli version. 

> **Analysis**: Not all configured properties removed from a cmd is considered as breaking change. The properties listed in CMD_PROPERTY_REMOVE_BREAK_LIST below are thought to influence or interrupt current workflow and be breaking changes.

Reviewer: ```suggestion
**Analysis**: Not all configured properties removed from a cmd are considered breaking change. The properties listed in CMD_PROPERTY_REMOVE_BREAK_LIST below are thought to influence or interrupt current workflow and be breaking changes.
```

> For now, removed properties in cmd level do not cause breaking workflows.

Reviewer: ```suggestion
For now, removed properties in cmd level does not break workflow.
```
